src/java.base/share/classes/java/security/PEMEncoder.java

```diff
@@ -0,0 +1,348 @@
+                        "contain PrivateKey.");
+                }
+                yield buildKey(kp.getPrivate().getEncoded(),
+                    kp.getPublic().getEncoded());
+            }
+            case X509EncodedKeySpec x ->
+                buildKey(null, x.getEncoded());
+            case PKCS8EncodedKeySpec p ->
+                buildKey(p.getEncoded(), null);
+            case EncryptedPrivateKeyInfo epki -> {
+                try {
+                    yield Pem.pemEncoded(Pem.ENCRYPTED_PRIVATE_KEY,
+                        epki.getEncoded());
+                } catch (IOException e) {
+                    throw new IllegalArgumentException(e);
+                }
+            }
+            case X509Certificate c -> {
+                try {
+                    if (isEncrypted()) {
+                        throw new IllegalArgumentException("Certificates " +
+                            "cannot be encrypted");
+                    }
+                    yield Pem.pemEncoded(Pem.CERTIFICATE, c.getEncoded());
+                } catch (CertificateEncodingException e) {
+                    throw new IllegalArgumentException(e);
+                }
+            }
+            case X509CRL crl -> {
+                try {
+                    if (isEncrypted()) {
+                        throw new IllegalArgumentException("CRLs cannot be " +
+                            "encrypted");
+                    }
+                    yield Pem.pemEncoded(Pem.X509_CRL, crl.getEncoded());
+                } catch (CRLException e) {
+                    throw new IllegalArgumentException(e);
+                }
+            }
+            case PEMRecord rec -> Pem.pemEncoded(rec);
+
+            default -> throw new IllegalArgumentException("PEM does not " +
+                "support " + de.getClass().getCanonicalName());
+        };
+    }
+
+    /**
+     * Encodes the specified {@code DEREncodable} into PEM.
+     *
+     * @param de the {@code DEREncodable} to be encoded.
+     * @return PEM encoded byte array
+     * @throws IllegalArgumentException when encoding the {@code DEREncodable}
+     * fails.
+     * @throws NullPointerException if {@code de} is {@code null}.
+     * @see #withEncryption(char[])
+     */
+    public byte[] encode(DEREncodable de) {
+        return encodeToString(de).getBytes(StandardCharsets.ISO_8859_1);
+    }
+
+    /**
+     * Returns a new {@code PEMEncoder} instance configured with the default
+     * encryption algorithm and a given password.
+     *
+     * <p> Only {@link PrivateKey} objects can be encrypted with this newly
+     * configured instance.  Encoding other {@link DEREncodable} objects will
+     * throw an{@code IllegalArgumentException}.
+     *
+     * @implNote The default algorithm is defined by Security Property {@code
+     * jdk.epkcs8.defaultAlgorithm} using default password-based encryption
+     * parameters by the supporting provider.  To configure all the encryption
```

Comment:
Maybe reword as "If you need more control over the encryption algorithm and parameters, use the ..."